### PR TITLE
Add Time Trigger API Support for nidaqmx-python

### PR DESCRIPTION
**Pull request by charitylxy**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?
This PR adds Time Trigger API support for the following functions in nidaqmx-python:
[DAQmxCfgTimeStartTrig ](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/daqmx/api/nicai/source/nicai/ctrigger.cpp&version=GBmain&line=209&lineEnd=210&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)

Remaining API yet to be added (due to functions missing in daqmxAPISharp.json): 
[DAQmxCfgAnlgMultiEdgeRefTrig](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/daqmx/api/nicai/source/nicai/ctrigger.cpp&version=GBmain&line=389&lineEnd=436&lineStartColumn=1&lineEndColumn=2&lineStyle=plain&_a=contents)
[DAQmxCfgAnlgMultiEdgeStartTrig](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/daqmx/api/nicai/source/nicai/ctrigger.cpp&version=GBmain&line=121&lineEnd=166&lineStartColumn=1&lineEndColumn=2&lineStyle=plain&_a=contents)

Work Item: [[DAQmx] Add Time Triggers API support for nidaqmx-python](https://ni.visualstudio.com/DevCentral/_workitems/edit/2610992)

### Why should this Pull Request be merged?
It provide support for Time Trigger in python.

### What testing has been done?
- Ran nidaqmx-python code generator and validated the API was generated correctly
- TODO: Regression tests for Time Trigger API to be added

### CODE REVIEW

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py [hunk 1]
```diff
@@ -1037,6 +1037,20 @@ def cfg_dig_pattern_start_trig(
         self._interpreter.cfg_dig_pattern_start_trig(
             self._handle, trigger_source, trigger_pattern, trigger_when.value)
 
+    def cfg_time_start_trig(self, when, timescale=Timescale.USE_HOST):
+        """
+        New Start Trigger
+
+        Args:
+            when (nidaqmx.constants.DateTime): Specifies when to
```
Comment (zhindes) [inline generated/nidaqmx/_task_modules/triggering/start_trigger.py:1045]: I don't think this is the right type. I added [AbsoluteTime in _lib_time.py](https://github.com/ni/nidaqmx-python/blob/c5fddf5c9dd6be9b0d02bcd96338963176515cf3/src/handwritten/_lib_time.py) that we should accept. We should also accept a `datetime` and a `hightime` from the std library.

File: generated/nidaqmx/_library_interpreter.py [hunk 2]
```diff
@@ -390,6 +390,18 @@ def cfg_samp_clk_timing(
             task, source, rate, active_edge, sample_mode, samps_per_chan)
         self.check_for_error(error_code)
 
+    def cfg_time_start_trig(self, task, when, timescale):
+        cfunc = lib_importer.windll.DAQmxCfgTimeStartTrig
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        lib_importer.task_handle, DateTime, ctypes.c_int]
```
Comment (zhindes) [inline generated/nidaqmx/_library_interpreter.py:399]: When should be an `AbsoluteTime`. `AbsoluteTime` is a ctypes type that should work correctly here, I think.

File: generated/nidaqmx/_grpc_interpreter.py [hunk 3]
```diff
@@ -378,6 +378,11 @@ def cfg_samp_clk_timing(
                 active_edge_raw=active_edge, sample_mode_raw=sample_mode,
                 samps_per_chan=samps_per_chan))
 
+    def cfg_time_start_trig(self, task, when, timescale):
+        response = self._invoke(
+            self._client.CfgTimeStartTrig,
+            grpc_types.CfgTimeStartTrigRequest(task=task, when=when, timescale_raw=timescale))
```
Comment (zhindes) [inline generated/nidaqmx/_grpc_interpreter.py:384]: what type does the grpc lib use for when?

**Review by zhindes — CHANGES_REQUESTED** [review]
I think the types aren't correct here. Some tests would have found a lot of issues, I think?

File: generated/nidaqmx/_grpc_interpreter.py (hunk 3 above)
Comment (bkeryan) [reply generated/nidaqmx/_grpc_interpreter.py:384]: google.protobuf.Timestamp
https://github.com/ni/nidaqmx-python/blob/master/src/codegen/protos/nidaqmx.proto#L4690
https://github.com/ni/nidaqmx-python/blob/master/src/handwritten/_grpc_time.py

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py (hunk 1 above)
Comment (bkeryan) [reply generated/nidaqmx/_task_modules/triggering/start_trigger.py:1045]: The interpreter interface should only use datetime/hightime, not the ctypes and gRPC specific types.

The interpreter implementations should be responsible for converting datetime/hightime to/from AbsoluteTime/google.protobuf.Timestamp.

Comment (zhindes) [reply generated/nidaqmx/_task_modules/triggering/start_trigger.py:1045]: Yup, you're right. I'm just remembering everything we talked about when I added `AbsoluteTime`. `AbsoluteTime` is the type that the lib_interpreter uses to interface with the CAPI. `google.protobuf.Timestamp` for the grpc_interpreter. In addition, the top-level API should accept `datetime` or `hightime` as well. So just pass them down to the interpreter and each will convert to their respective types.

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py
Comment (bkeryan) [inline generated/nidaqmx/_task_modules/triggering/start_trigger.py:1]: In addition to this config function, several time trigger attributes are missing from [start_trigger.py](https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/triggering/start_trigger.py) and similar classes. Are these on your radar?

https://www.ni.com/docs/en-US/bundle/ni-daqmx-properties/page/daqmxprop/daqmxtrigger.html
- Start:Time:When
- Start:Timestamp:Value
- Reference:Timestamp:Value
- More:Arm Start:Time:When
- More:Arm Start:Timestamp:Value

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py (hunk 1 above)
Comment (bkeryan) [reply generated/nidaqmx/_task_modules/triggering/start_trigger.py:1045]: Some starting points:
- Library
  - https://github.com/ni/nidaqmx-python/blob/master/src/codegen/templates/library_interpreter/default_c_function_call.py.mako
  - https://github.com/ni/nidaqmx-python/blob/master/src/codegen/utilities/interpreter_helpers.py#L135
- gRPC
  - https://github.com/ni/nidaqmx-python/blob/master/src/codegen/templates/grpc_interpreter/default_grpc_function_call.py.mako
  - https://github.com/ni/nidaqmx-python/blob/master/src/codegen/utilities/interpreter_helpers.py#L329

File: generated/nidaqmx/_task_modules/triggering/start_trigger.py
Comment (charitylxy) [reply generated/nidaqmx/_task_modules/triggering/start_trigger.py:1]: Opps, I might've missed the attributes out, will look into adding those.

**Pull request closed without merging**